src/main/java/teammates/storage/search/SearchQuery.java

```diff
@@ -40,14 +43,31 @@ public int getFilterSize(){
     }
     
     protected SearchQuery setTextFilter(String textField, String queryString){
-        String sanitizedQueryString = Sanitizer.sanitizeForSearch(queryString).toLowerCase().trim();
+        String sanitizedQueryString;
+        if(isValidEmailAddress(queryString)){
+            sanitizedQueryString = queryString.toLowerCase().trim();
+        } else {
+            sanitizedQueryString = Sanitizer.sanitizeForSearch(queryString).toLowerCase().trim(); 
+        }
+        
         if(!sanitizedQueryString.isEmpty()){
             String preparedOrQueryString = prepareOrQueryString(sanitizedQueryString);
             this.textQueryStrings.add(textField + ":" + preparedOrQueryString);
         }
         return this;
     }
     
+    private boolean isValidEmailAddress(String email) {
```

Comment:
Thanks for the detailed review comments @damithc. Utilized already available regex for the validation. Instead of using it directly in the `setTextFilter` method, called it via the `isValidEmailAddress` method thinking the code would be more clearer.